### PR TITLE
Document.modify() support

### PR DESCRIPTION
**Pull request by thomasst**
DocumentBase/SoftDeleteDocument: implement modify() in addition to update()

Before merge:
- [x] Merge https://github.com/closeio/mongoengine/pull/14 first, then update `requirements.txt

### CODE REVIEW

**Comment by thomasst** [comment]
@wojcikstefan please review this one as well.

File: flask_common/documents.py [hunk 1]
```diff
@@ -75,6 +75,12 @@ def save(self, *args, **kwargs):
             self.date_updated = now
         return super(DocumentBase, self).save(*args, **kwargs)
 
+    def modify(self, *args, **kwargs):
```
Comment (wojcikstefan) [inline flask_common/documents.py]: If `upsert` is True, this should also set `date_created`. I'm fine with also throwing `NotImplementedError` for upserts though.

**Comment by wojcikstefan** [comment]
LGTM

File: flask_common/documents.py (hunk 1 above)
Comment (thomasst) [inline flask_common/documents.py]: If the document doesn't have a pk, we throw `InvalidDocumentError` in `Document.modify()`, so I don't think this is ever a valid scenario.

**Pull request merged**